app/components/channel_list/categories/categories.tsx

```diff
@@ -19,20 +19,26 @@ const styles = StyleSheet.create({
     },
 });
 
-const renderCategory = (data: {item: TempoCategory}) => (
-    <>
-        <CategoryHeader heading={data.item.title}/>
-        <CategoryBody channels={data.item.channels}/>
-    </>
-);
+const renderCategory = (data: {item: CategoryModel}) => {
+    return (
+        <>
+            <CategoryHeader category={data.item}/>
+            <CategoryBody category={data.item}/>
+        </>
+    );
+};
 
 const Categories = (props: Props) => {
+    // Sort Categories
+    props.categories.sort((a, b) => a.sortOrder - b.sortOrder);
```

Comment:
This sorting will happen each time this component re-renders ?  If we have a huge list that can cause some issue.  You could useMemo here.

Reply:
Also, why do we need it ?

Reply:
We need the re-render; the category sort order can change.

There are list-view optimisations coming in the next PR that will help with performance of larger lists.